src/rgw/rgw_rados.cc

```diff
@@ -8210,7 +8213,15 @@ int RGWRados::fetch_remote_obj(RGWObjectCtx& obj_ctx,
   return 0;
 set_err_state:
   if (copy_if_newer && ret == -ERR_NOT_MODIFIED) {
-    ret = 0;
+    if (olh_epoch && *olh_epoch > 0) {
+      // we may have already fetched during sync of OP_ADD, but were waiting
+      // for OP_LINK_OLH to call set_olh() with a real olh_epoch
+      ret = set_olh(obj_ctx, dest_bucket_info, dest_obj, false, nullptr,
+                    *olh_epoch, real_time(), false, zones_trace);
```

Comment:
this call was not updating the data changes log, so was causing checkpoint failures in testing

Reply:
Ouch. Great work finding it.